src/java.base/share/classes/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.java

```diff
@@ -68,6 +69,22 @@
  *   // ... and so on
  * }}</pre>
  *
+ * However, it is preferable to use {@link VarHandle}:
+ * <pre> {@code
+ * class Node {
+ *   private volatile Node left, right;
+ *
+ *   private static final VarHandle LEFT = MethodHandles.lookup().
+ *      findVarHandle(Node.class, "left", Node.class);
+ *   private static final VarHandle RIGHT = MethodHandles.lookup().
+ *      findVarHandle(Node.class, "right", Node.class);
+ *   Node getLeft() { return left; }
+ *   boolean compareAndSetLeft(Node expect, Node update) {
+ *     return LEFT.compareAndSet(this, expect, update);
+ *   }
+ *   // ... and so on
+ * }}</pre>
```

Comment:
I wonder if this should be convert to a `{@snippet ..}` and have it handle the checked exceptions thrown by findVarHandle. I'm just thinking of someone seeing VarHandle usage for first time, then reporting a bug that the example doesn't compile.